### PR TITLE
Remove usages of & on references

### PR DESCRIPTION
**Pull request by pleroy**
Addresses #432.

### CODE REVIEW

File: ksp_plugin/plugin.cpp [hunk 1]
```diff
@@ -311,7 +311,7 @@ void Plugin::InsertCelestial(
                       << "of initialization";
   auto const it = celestials_.find(parent_index);
   CHECK(it != celestials_.end()) << "No body at index " << parent_index;
-  Celestial const& parent= *it->second;
+  not_null<Celestial const*> parent= it->second.get();
```
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]:
```
/=
i 
```

File: ksp_plugin/plugin.cpp [hunk 2]
```diff
@@ -498,17 +498,14 @@ RenderedTrajectory<World> Plugin::RenderedVesselTrajectory(
...
-       !apparent_it.at_end();
-       ++apparent_it) {
-    final_state = &apparent_it.degrees_of_freedom();
-    if (initial_state != nullptr) {
-      result.emplace_back(to_world(initial_state->position()),
-                          to_world(final_state->position()));
+  auto initial_it = apparent_trajectory->first();
+  if (!initial_it.at_end()) {
+    auto final_it = initial_it;
+    for (; ++final_it, !final_it.at_end();) {
```
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: please make this loop even more screwed-up.

File: ksp_plugin/plugin.cpp (hunk 1 above)
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Done.

**Pull request merged**